src/components/DatasetPage/DatasetPage.tsx

```diff
@@ -26,52 +30,223 @@ export function DatasetPage() {
       })
   }, [])
 
+  const editButton = () => {
+    setEdit((value) => !value)
+  }
+
+  const onFinish = (values: any) => {
+    setEdit((value) => !value)
+    const editedDataset = objectToString(values)
+    editDataset(datasetName, editedDataset, userContext)
+      .then(() => {
+        SuccessNotification({
+          type: 'success',
+          message: 'Dataset updated',
+        })
+      })
+      .catch((error) => {
+        setIsLoading(false), setError(error)
+      })
+  }
+
   return isLoading ? (
     <Loading />
   ) : (
     <>
       {!error && (
         <>
-          <Title>Dataset {dataset.name}</Title>
+          <Title>Dataset {datasetName}</Title>
           <Card>
-            <Descriptions bordered column={3} labelStyle={{ fontWeight: 'bold' }} size="small">
-              <Descriptions.Item label="Fetal fraction preface">
-                {dataset.fetal_fraction_preface}
-              </Descriptions.Item>
-              <Descriptions.Item label="Fetal fraction y for trisomy">
-                {dataset.fetal_fraction_y_for_trisomy}
-              </Descriptions.Item>
-              <Descriptions.Item label="Fetal fraction y max">
-                {dataset.fetal_fraction_y_max}
-              </Descriptions.Item>
-              <Descriptions.Item label="Fetal fraction y min">
-                {dataset.fetal_fraction_y_min}
-              </Descriptions.Item>
-              <Descriptions.Item label="Fetal fraction XXX">
-                {dataset.fetal_fraction_XXX}
-              </Descriptions.Item>
-              <Descriptions.Item label="Fetal fraction X0">
-                {dataset.fetal_fraction_X0}
-              </Descriptions.Item>
-              <Descriptions.Item label="Y axis min">{dataset.y_axis_min}</Descriptions.Item>
-              <Descriptions.Item label="Y axis max">{dataset.y_axis_max}</Descriptions.Item>
-              <Descriptions.Item label="K upper">{dataset.k_upper}</Descriptions.Item>
-              <Descriptions.Item label="K lower">{dataset.k_lower}</Descriptions.Item>
-              <Descriptions.Item label="M lower">{dataset.m_lower}</Descriptions.Item>
-              <Descriptions.Item label="M upper">{dataset.m_upper}</Descriptions.Item>
-              <Descriptions.Item label="Trisomy soft max">
-                {dataset.trisomy_soft_max}
-              </Descriptions.Item>
-              <Descriptions.Item label="Trisomy hard max">
-                {dataset.trisomy_hard_max}
-              </Descriptions.Item>
-              <Descriptions.Item label="Trisomy hard min">
-                {dataset.trisomy_hard_min}
-              </Descriptions.Item>
-              <Descriptions.Item label="Comment" span={3}>
-                {dataset.comment}
-              </Descriptions.Item>
-            </Descriptions>
+            <Form
```

Comment:
You don't need to assign each entry one by one. This whole thing can become:  `<Form layout="inline" onFinish={onFinish} form={form} initialValues={{ ...dataset }}>`